vm/error.go

```diff
@@ -1,18 +1,68 @@
 package vm
 
-type ErrorClass struct {
-	*BaseClass
+import (
+	"fmt"
+)
+
+var (
+	// UndefinedMethodErrorClass ...
+	UndefinedMethodErrorClass *RClass
+	// ArgumentErrorClass ...
+	ArgumentErrorClass *RClass
+	// TypeErrorClass ...
+	TypeErrorClass *RClass
+)
+
+const (
+	// UndefinedMethodError describes the error type in string
+	UndefinedMethodError = "UndefinedMethodError"
+	// ArgumentError describes the error type in string
+	ArgumentError = "ArgumentError"
+	// TypeError describes the error type in string
+	TypeError = "TypeError"
+)
+
+func init() {
+	bc := createBaseClass(UndefinedMethodError)
+	UndefinedMethodErrorClass = &RClass{BaseClass: bc}
+	bc = createBaseClass(ArgumentError)
+	ArgumentErrorClass = &RClass{BaseClass: bc}
+	bc = createBaseClass(TypeError)
+	TypeErrorClass = &RClass{BaseClass: bc}
 }
 
+// Error ...
 type Error struct {
-	Class   *ErrorClass
+	Class   *RClass
 	Message string
 }
 
+// Inspect ...
 func (e *Error) Inspect() string {
 	return "ERROR: " + e.Message
 }
 
 func (e *Error) returnClass() Class {
 	return e.Class
 }
+
+func initializeUndefinedMethodError(format string, args ...interface{}) *Error {
```

Comment:
Since we don't want to create struct for every type of error, I think we also shouldn't create a function for every error type.

I would write something like:

```go
func initializeError(errorType *RClass, format string, args ...interface{}) *Error {
    return &Error{
        Class: errorType,
        Message: fmt.Sprintf(errorType.class.Name + ": "+format, args...),
    }
}

And this can be used on any types of error.